src/communication/bolt/v1/states/handlers.hpp

```diff
@@ -96,6 +96,10 @@ State HandleRun(TSession &session, const State state, const Value &query, const
     vec.reserve(header.size());
     for (auto &i : header) vec.emplace_back(std::move(i));
     data.emplace("fields", std::move(vec));
+    if (return_qid && qid.has_value()) {
+      data.emplace("qid", Value{*qid});
+    }
+
```

Comment:
Until now this implementation of run handlers was the same but now that the `qid` was introduced I would rather separate implementations, and remove this general one (even though they will mostly be the same, this is a specific case where it makes sense to have different methods which are very similar but should be multiple of them since they are implemented for different Bolt versions)